apps/calendar/js/time_observer.js

```diff
@@ -0,0 +1,111 @@
+define(function(require, exports) {
+'use strict';
+
+// this module abstracts minute and day changes, it should be used by all the
+// views that needs to listen for time changes (ie. update current time/date).
+// listeners are enabled/disabled when the document visibility changes,
+// reducing battery usage and making sure we always display the proper value.
+// ---
+// XXX: we don't listen to moztimechange here because the app forces a restart
+// when the time changes; even if the app did not restart we would still need
+// to trigger a refresh of most of the views (timezone affects the way we
+// busytimes are rendered in all the views, even the view/edit event views)
+// see: https://bugzilla.mozilla.org/show_bug.cgi?id=1093016#c9
+
+var EventEmitter2 = require('ext/eventemitter2');
+var EVENT = 'change';
+
+function TimeSignal(checkDiff) {
+  this._emitter = new EventEmitter2();
+  this._timeout = null;
+  this._prevTick = null;
+  this._checkDiff = checkDiff;
+  this._tick = this._tick.bind(this);
+}
+
+TimeSignal.prototype = {
+  add: function(cb) {
+    this._emitter.on(EVENT, cb);
+    this._start();
```

Comment:
very few event emitters support the `newListener` event and I don't think the EventEmitter2 emits a special event when removing listeners.. we would still need to override the `on/off` to keep the current functionality..

TBH, I would use signals for almost all the events since in my opinion the [benefits are greater than the disadvantages](https://github.com/millermedeiros/js-signals/wiki/Comparison-between-different-Observer-Pattern-implementations#signals) but I'm fine with any approach, [I don't care what color the bike shed is](http://bikeshed.com/).